core/server/proxy/src/main/java/alluxio/proxy/s3/signature/AwsCredential.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3.signature;
+
+import static alluxio.proxy.s3.signature.SignerConstants.DATE_FORMATTER;
+
+import alluxio.proxy.s3.S3Exception;
+import alluxio.proxy.s3.S3ErrorCode;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+
+/**
+ * Credential in the AWS authorization header.
+ * Ref: https://docs.aws.amazon.com/AmazonS3/latest/API/
+ * sigv4-auth-using-authorization-header.html
+ *
+ */
+public class AwsCredential {
+  private static final Logger LOG = LoggerFactory.getLogger(AwsCredential.class);
+
+  private final String mAccessKeyID;
+  private final String mDate;
+  private final String mAwsRegion;
+  private final String mAwsService;
+  private final String mAwsRequest;
+
+  AwsCredential(String accessKeyID, String date, String awsRegion,
+                String awsService, String awsRequest) {
+    mAccessKeyID = accessKeyID;
+    mDate = date;
+    mAwsRegion = awsRegion;
+    mAwsService = awsService;
+    mAwsRequest = awsRequest;
+  }
+
+  /**
+   * Parse credential value.
+   *
+   * Sample credential value:
+   * Credential=testuser/20220316/us-east-1/s3/aws4_request
+   *
+   * @throws S3Exception
+   */
+
+  /**
+   * @return AccessKeyID
+   */
+  public String getAccessKeyID() {
+    return mAccessKeyID;
+  }
+
+  /**
+   * @return date
+   */
+  public String getDate() {
+    return mDate;
+  }
+
+  /**
+   * @return region
+   */
+  public String getAwsRegion() {
+    return mAwsRegion;
+  }
+
+  /**
+   * @return service name
+   */
+  public String getAwsService() {
+    return mAwsService;
+  }
+
+  /**
+   * @return request info
+   */
+  public String getAwsRequest() {
+    return mAwsRequest;
+  }
+
+  /**
+   * @return formatted scope string
+   */
+  public String createScope() {
+    return String.format("%s/%s/%s/%s", getDate(),
+       getAwsRegion(), getAwsService(),
+       getAwsRequest());
+  }
+
+  /**
+   * Factory to create a {@link AwsCredential} instance.
+   */
+  public static class Factory {
```

Comment:
This Factory class doesn't seem to be necessary. You can just put the static `create` method in the `AwsCredential` class itself.